systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java

```diff
@@ -453,6 +458,178 @@ void testAddingAndRemovingJbodVolumes() {
+        if (Environment.isKafkaNodePoolsEnabled()) {
+            KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getBrokerPoolName(), kafkaNodePool -> {
+                JbodStorage storage = (JbodStorage) kafkaNodePool.getSpec().getStorage();
+                storage.getVolumes().remove(metadataVol);
+            }, testStorage.getNamespaceName());
+        } else {
+            KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> {
+                JbodStorage storage = (JbodStorage) kafka.getSpec().getKafka().getStorage();
+                storage.getVolumes().remove(metadataVol);
+            }, testStorage.getNamespaceName());
+        }
+
+        // Wait util it rolls
+        brokerPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getBrokerSelector(), 3, brokerPods);
+
+        // verify that Kraft metadata log will be re-assigned to another volume (the minimum id, which is 0 now that's why data-0)
+        verifyKafkaKraftMetadataLog(testStorage, 0, 2);
+
+        resourceManager.createResourceWithWait(clients.consumerStrimzi());
+        ClientUtils.waitForInstantConsumerClientSuccess(testStorage);
+
+        // ##############################
+        // Validate that continuous clients finished successfully
+        // ##############################
+        ClientUtils.waitForContinuousClientSuccess(testStorage, continuousClientsMessageCount);
+        // ##############################
+    }
+
+    /**
+     * Verifies the presence or absence of KRaft metadata logs across specified volume directories within Kafka pods.
+     * This method iterates through all Kafka pods retrieved based on the broker selector from the test storage configuration,
+     * checking each configured volume for the presence of KRaft metadata files. The method asserts that the metadata log
+     * exists only in the specified KRaft metadata volume and not in others, ensuring correct metadata log placement
+     * according to the test configuration.
+     *
+     * @param testStorage               An instance of TestStorage containing configuration and context for the current test,
+     *                                  including namespace and broker selector for identifying relevant Kafka pods.
+     * @param kraftMetadataVolumeId     The volume ID expected to contain the KRaft metadata log. This method will
+     *                                  assert the presence of metadata logs in this volume and their absence in others.
+     * @param numberOfVolumes           The total number of volumes configured in the JBOD (Just a Bunch Of Disks) for each Kafka broker.
+     *                                  This dictates how many volume directories the method will check within each Kafka pod.
+     */
+    private void verifyKafkaKraftMetadataLog(final TestStorage testStorage, final int kraftMetadataVolumeId, final int numberOfVolumes) {
```

Comment:
This could be useful for other tests maybe as well or? At least for the migration stuff or something. So I would move it to some `KafkaUtils` or something like that

Reply:
Sure I can move it there